src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -172,8 +178,19 @@ func (b *ByocAws) deploy(ctx context.Context, req *defangv1.DeployRequest, cmd s
 		return nil, errors.New("maximum number of services reached")
 	}
 
+	quotas := quota.NewAwsServiceQuotas(ctx, cfg)
+	if err := quotas.Initialize(ctx, cfg); err != nil {
+		term.Warnf("failed to initialize quotas: %v", err)
+	}
+
 	serviceInfos := []*defangv1.ServiceInfo{}
 	for _, service := range project.Services {
+		if quotas != nil {
+			if err := quotas.ValidateGpuQuota(ctx, &service); err != nil {
```

Comment:
you're validating each service one by one but what if the sum of all GPU's is too large?